src/base/bittorrent/torrentcreatorthread.cpp

```diff
@@ -178,17 +176,19 @@ void TorrentCreatorThread::run()
                 newTorrent.add_tracker(tracker.trimmed().toStdString(), tier);
         }
 
-        if (isInterruptionRequested()) return;
-
         // calculate the hash for all pieces
         lt::set_piece_hashes(newTorrent, Utils::Fs::toNativePath(parentPath).toStdString()
             , [this, &newTorrent](const lt::piece_index_t n)
         {
+            if (isInterruptionRequested())
+                throw RuntimeError {tr("Create new torrent aborted.")};
```

Comment:
Isn't this callback executed in another thread? Do you really able to handle this exception?

Reply:
It was suggested here: https://github.com/arvidn/libtorrent/issues/6024#issuecomment-789757211
And it works fine in my testing.

Reply:
>And it works fine in my testing.

Ok 👌